Kuroba/app/src/main/java/com/github/adamantcheese/chan/ui/controller/WatchSettingsController.java

```diff
@@ -84,6 +85,14 @@ public void onCreate() {
     @Override
     public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
         ChanSettings.watchEnabled.set(isChecked);
+
+        if (!isChecked) {
+            ChanSettings.watchBackground.setSync(false);
+            ((StartActivity) context).restartApp();
```

Comment:
This is not the correct way of doing this, no way in hell. The EventBus message sent by the background setting being toggled should take care of removing the watch manager from the wake manager's listener set, which would stop background updates. Restarting the app when it is toggled off is horrible for user experience.

Reply:
Oh, damn. The main problem for me to choose this approach to solve it was that when turning watcher off in the settings and then automatically turning the background watcher off too the view didn't want to get updated for some reason. I tried couple of methods but they didn't work so I thought that this one should work, but apparently didn't notice that this may be called not only when you change the setting by yourself but also in some other cases. I will fix it.

Reply:
This still wasn't correct, as you were dealing with the issue too far away from the actual problem. I've merged this and fixed it after the fact in WatchManager when it's checking the currentInterval and newInterval (which I changed because the naming convention didn't make sense). Please take a look at it there or in https://github.com/Adamantcheese/Kuroba/commit/8389385bc165acafa8cc5da79db5f75e3cbbc5e1.